File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
from heapq import heappush, heappop, heapify

def solution(scoville, K):
    answer = 0
    
    heapify(scoville)
    
    while(scoville[0] < K):
        
        a = heappop(scoville)
        b = heappop(scoville)
        
        heappush(scoville, a + 2*b)
        answer += 1
        
        if len(scoville) == 1 and scoville[0] < K:
            answer = -1
            break
        
    return answer

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int i = 0; i < scoville.length; i++){
            pq.add(scoville[i]);
        }
        
        while(pq.peek() < K){
            if (pq.size() == 1){
                return -1;
            }
            int a = pq.poll();
            int b = pq.poll();
            
            pq.add(a + 2 * b);
            answer += 1;
               
        }
        
        return answer;
        
        
    }
}